File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */

var md = 0;
var deleteMiddle = function(head) {
    // md=0;
    return fun(head,0);
};

let fun = (ptr,n)=>{
    if(ptr===null){
        md = Math.floor(n/2);
        return null;
    }
    ptr.next=fun(ptr.next,n+1);
         // console.log(md);
    if(n==md){
        return ptr.next;
    }
    return ptr;
}

File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int md = 0;
    ListNode* deleteMiddle(ListNode* head) {
        return fun(head,0);
    }
    ListNode* fun(ListNode* ptr,int n){
        if(ptr==nullptr){
            md = (n)/2;
            return nullptr;
        }
        ptr->next = fun(ptr->next,n+1);
        if(n==md){
            return ptr->next;
        }
        return ptr;
    }
};